### PR TITLE
Sort index on extend

### LINKED ISSUE #19: consider sorting the index in extend
For many dask operations it very useful to have the index in each partition sorted. In certain cases, particularly time indexed frames, operations like resample break without a sorted index in each partition. I think we should do this either by default or have an option to do this

### PR DESCRIPTION
**Pull request by mrocklin**
Fixes #19

### CODE REVIEW

File: castra/core.py [hunk 1]
```diff
@@ -139,6 +139,7 @@ def load_categories(self, loads=pickle.loads):
 
     def extend(self, df):
         # TODO: Ensure that df is consistent with existing data
+        df = df.sort_index(inplace=False)
```
Comment (cpcloud) [inline castra/core.py]: no need for the parameter but no harm in passing it

Comment (cpcloud) [inline castra/core.py]: worth checking if it's already sorted with `df.index.is_monotonic_increasing`?

Comment (mrocklin) [inline castra/core.py]: fixed

**Comment by cpcloud** [comment]
LGTM

**Comment by cpcloud** [comment]
ok 2 merge?

**Pull request merged**